pydoc/index.rst

```diff
@@ -0,0 +1,20 @@
+DLite Python API
```

Comment:
This should be updated to be a TOC that is divides into 1) C API 2) Python API 3) Fortran API.
I think at least that there's a Fortran API as well?

This point can be made into a separate issue and updated later.

Reply:
Follow-up task created: #368 Extend Sphinx-generated documentation to include C/C++ and Fortran API